editoast/src/generated_data/error/mod.rs

```diff
@@ -14,17 +14,69 @@ use diesel::result::Error as DieselError;
 use diesel::PgConnection;
 
 use super::GeneratedData;
-use crate::infra_cache::InfraCache;
+use crate::infra_cache::{InfraCache, ObjectCache};
+use crate::schema::{InfraError, ObjectType};
 use graph::Graph;
 
+type ErrGenerators<'a> = [ErrGenerator];
+type FnErrGenerator = fn(&ObjectCache, &InfraCache, &Graph) -> Vec<InfraError>;
+
+pub struct ErrGenerator {
+    priority: u32,
+    check_function: FnErrGenerator,
+}
+impl ErrGenerator {
+    pub const fn new(priority: u32, check_function: FnErrGenerator) -> Self {
+        ErrGenerator {
+            priority,
+            check_function,
+        }
+    }
+    fn get(
+        &'static self,
+        object_type: &ObjectCache,
+        infra_cache: &InfraCache,
+        graph: &Graph,
+    ) -> Vec<InfraError> {
+        (self.check_function)(object_type, infra_cache, graph)
+    }
+}
+
+pub fn generate_errors(
+    object_type: ObjectType,
+    infra_cache: &InfraCache,
+    graph: &Graph,
+    list_err_generators: &'static ErrGenerators,
+) -> Vec<InfraError> {
+    let mut errors = Vec::new();
+    let mut current_priority = list_err_generators[0].priority;
+    for el in infra_cache.get_objects_by_type(object_type).values() {
+        let mut infra_errors = Vec::new();
+        for f in list_err_generators.iter() {
+            if infra_errors.is_empty() && f.priority == current_priority {
+                infra_errors.extend(f.get(el, infra_cache, graph));
+            } else if infra_errors.is_empty() && f.priority != current_priority {
+                current_priority = f.priority;
+                infra_errors.extend(f.get(el, infra_cache, graph));
+            } else if !infra_errors.is_empty() && current_priority == f.priority {
+                infra_errors.extend(f.get(el, infra_cache, graph));
+            } else {
+                break;
+            }
+        }
+        errors.extend(infra_errors);
+    }
+    errors
```

Comment:
The code is not clear and there is an error (current_priority should be reset to initial value for each object.

```suggestion
    let mut errors = Vec::new();
    for el in infra_cache.get_objects_by_type(object_type).values() {
        let mut found_error = false;
        let mut current_priority = list_err_generators[0].priority;
        for f in list_err_generators.iter() {
            if f.priority != current_priority && found_error {
                // No need to check further errors
                break;
            }
            let new_errors = (f.check_function)(el, infra_cache, graph);
            // Update found error
            found_error |= !new_errors.is_empty();
            // Add errors to the list
            errors.extend(new_errors);
            // Update priority
            current_priority = f.priority;
        }
    }
    errors
```